driver.py

```diff
@@ -22,9 +24,27 @@
 class SmartSimScalingTests:
 
     def __init__(self):
-        self.resnet_model = "./imagenet/resnet50.pt"
         self.date = str(datetime.datetime.now().strftime("%Y-%m-%d"))
 
+    def set_resnet_model(self, device="GPU"):
+        self.resnet_model = f"./imagenet/resnet50.{device}.pt"
+        if not Path(self.resnet_model).exists():
+            logger.info(f"AI Model {self.resnet_model} does not exist, it will be created")
+            save_model(device)
```

Comment:
I would suggest adding a force parameter to ensure that the model is saved off regardless.

Also, if I'm not mistaken, the driver script is run from the front-end, but the model saver has to be run in a back-end node. Do you need to adjust the driver to enforce this?

Reply:
Yeah, I'll add the `force` parameter. As for the back-end node, it is true you need it for GPU, but that would mean we should run the process through the workload manager (for CPU, it should make less difference, although there could be optimizations I'm not aware of, or you could be running on different architectures). In those cases, I wrote in README that the user should run the script before executing the driver script.

Reply:
I think you need a device={CPU/GPU} parameter on the call to save_model()